AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.java

```diff
@@ -167,8 +174,73 @@ public boolean onQueryTextChange(String newText) {
                 return true;
             }
         });
+
+        MenuItem addDecks = mToolbar.getMenu().findItem(R.id.deck_picker_dialog_action_add_deck);
+        addDecks.setOnMenuItemClickListener(menuItem -> {
+            // creating new deck without any parent deck
+            addDeckDialog("");
+            return true;
+        });
     }
 
+    private void addDeckDialog(String parentDeckPath) {
+        EditText dialogEditText = new FixedEditText(requireActivity());
+        dialogEditText.setSingleLine(true);
+        new MaterialDialog.Builder(requireActivity())
+                .title(R.string.new_deck)
+                .positiveText(R.string.dialog_ok)
+                .customView(dialogEditText, true)
+                .onPositive((dialog, which) -> {
+                    startCreatingDeck(parentDeckPath,dialogEditText.getText().toString());
+                })
+                .negativeText(R.string.dialog_cancel)
+                .show();
+    }
+
+    private void startCreatingDeck(String parentDeckPath, String deckName) {
+        String deckNameWithParentPath = parentDeckPath + deckName;
+        if (Decks.isValidDeckName(deckNameWithParentPath)) {
+            boolean creationSucceeded = createNewDeck(deckNameWithParentPath);
+            if (!creationSucceeded) {
+                UIUtils.showThemedToast(requireActivity(), getString(R.string.failed_creation_deck_name), false);
+            }
+        } else {
+            Timber.i("configureFloatingActionsMenu::addDeckButton::onPositiveListener - Not creating invalid deck name '%s'", deckNameWithParentPath);
+            UIUtils.showThemedToast(requireActivity(), getString(R.string.invalid_deck_name), false);
+        }
+    }
+
+    @NonNull
+    protected AnkiActivity requireAnkiActivity() {
+        return (AnkiActivity) requireActivity();
+    }
+
+    /**
+     * @return if the deck exists after the call (created, or already exists)
+     * no-op if deckName is already present in list
+     */
+    private boolean createNewDeck(@NonNull String deckName) {
+        try {
+            // handled if duplicate name deck is added
+            Decks decks = requireAnkiActivity().getCol().getDecks();
+            if (decks.byName(deckName) == null) {
+                Timber.i("DeckPicker:: Creating new deck...");
+                Long id = requireAnkiActivity().getCol().getDecks().id(deckName);
+                SelectableDeck dec = new SelectableDeck(id, deckName);
+                mAdapter.mCurrentlyDisplayedDecks.add(dec);
+                mAdapter.mAllDecksList.add(dec);
+                Collections.sort(mAdapter.mCurrentlyDisplayedDecks);
+                mAdapter.notifyDataSetChanged();
+            } else {
+                UIUtils.showThemedToast(requireActivity(), getString(R.string.duplicate_deck_name), false);
+                return false;
```

Comment:
See comment: `if the deck exists after the call (created, or already exists)`

```suggestion
                return true;
```